src/GraphQL.Tests/Extensions/GraphQLExtensionsTests.cs

```diff
@@ -4,20 +4,173 @@
+                yield return new object[] { new ListGraphType(new NonNullGraphType(new BooleanGraphType())), new object[] { true, false, null }, false };
+                yield return new object[] { new ListGraphType(new NonNullGraphType(new BooleanGraphType())), new object[] { true, false, true }, true };
+
+                yield return new object[] { new InputObjectGraphType<Person>(), null, true };
+                yield return new object[] { new NonNullGraphType(new InputObjectGraphType<Person>()), null, false };
+                yield return new object[] { new InputObjectGraphType<Person>(), new Person(), true };
+                yield return new object[] { new InputObjectGraphType<Person>(), "aaa", false };
+
+                // https://github.com/graphql-dotnet/graphql-dotnet/issues/2334
+                yield return new object[] { new ListGraphType(new BooleanGraphType()), true, true };
+            }
+
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        }
+
+        public class IsValidDefaultExceptionTestData : IEnumerable<object[]>
+        {
+            public IEnumerator<object[]> GetEnumerator()
+            {
+                yield return new object[] { null, 0, new ArgumentNullException("type") };
+                yield return new object[] { new ObjectGraphType(), 0, new ArgumentOutOfRangeException("type", "Must provide Input Type, cannot use ObjectGraphType 'Object'") };
+            }
+
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        }
+
+        public class ToASTTestData : IEnumerable<object[]>
+        {
+            public class Person
+            {
+                public int Age { get; set; }
+
+                public string Name { get; set; }
+            }
+
+            public class PersonInputType : InputObjectGraphType<Person>
+            {
+                public override IValue ToAST(object value)
+                {
+                    var person = (Person)value;
+
+                    return new ObjectValue(new[]
+                    {
+                        new ObjectField("Name", new StringValue(person.Name)),
+                        new ObjectField("Age", new IntValue(person.Age))
+                    });
+                }
+            }
+
+            public IEnumerator<object[]> GetEnumerator()
+            {
+                yield return new object[] { null, true, new NullValue() };
+
+                yield return new object[] { new BooleanGraphType(), true, new BooleanValue(true) };
+                yield return new object[] { new BooleanGraphType(), false, new BooleanValue(false) };
+                yield return new object[] { new BooleanGraphType(), null, new NullValue() };
+
+                yield return new object[] { new NonNullGraphType(new BooleanGraphType()), false, new BooleanValue(false) };
+                //yield return new object[] { new NonNullGraphType(new BooleanGraphType()), null, false }; //TODO: exception ?
```

Comment:
What about that case?

Reply:
It seems reasonable to throw an `InvalidOperationException` if you are trying to coerce a value that is invalid for the type.  In this case, trying to cast a null value to a non-null type, so-to-speak.  Not much different than trying to convert the string "abc" to a BooleanGraphType AST representation.